### PR TITLE
Unresolved Braches Cohort Selection Script

### PR DESCRIPTION
**Pull request by maxxcrawford**
See [Bug #1651817](https://bugzilla.mozilla.org/show_bug.cgi?id=1651817) for more information.

### CODE REVIEW

File: scripts/collect-unresolved-breaches-addresses.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  return false;
+}
+
+(async () => {
+  const allHibpBreaches = await HIBP.req("/breaches");
+
+  // "SELECT primary_email, primary_sha1 FROM subscribers WHERE signup_language LIKE 'en%' AND breaches_resolved IS NULL ORDER BY random();"
+  const query = knex("subscribers").where("signup_language", "like", "en%").andWhere({breaches_resolved: null}).orderByRaw("RANDOM()").select("primary_email", "primary_sha1");
```
Comment (maxxcrawford) [inline scripts/collect-unresolved-breaches-addresses.js:24]: Confirm `signup_language` has an index

**Comment by groovecoder** [comment]
I added a migration to add an index to `subscribers.signup_language`. Run locally:

`./node_modules/.bin/knex --knexfile db/knexfile.js migrate:latest`

Then make sure the script still runs.

After merge, we'll run it on dev server too.

File: scripts/collect-unresolved-breaches-addresses.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const Knex = require("knex");
+const knexConfig = require("../db/knexfile");
+const knex = Knex(knexConfig);
+
+const HIBP = require("../hibp");
+
+async function checkIfBreachesExist(sha1, breaches) {
+  const breachResults = await HIBP.getBreachesForEmail(sha1, breaches, true);
+
+  if (breachResults.length > 1) {
```
Comment (groovecoder) [inline scripts/collect-unresolved-breaches-addresses.js:14]:
```suggestion
  if (breachResults.length >= 1) {
```

File: scripts/collect-unresolved-breaches-addresses.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+function getArgsValue(argument) {
+  const cliArguments = process.argv;
+
+  if (cliArguments.indexOf(argument) < 0) {
+    throw new Error(`You are missing the argument: ${argument}`);
+  }
+
+  const arguemntIndex = cliArguments.indexOf(argument);
```
Comment (pdehaan) [inline scripts/collect-unresolved-breaches-addresses.js:28]: typo: "argumentIndex"

File: scripts/collect-unresolved-breaches-addresses.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  const valueNumber = parseInt(value);
+
+  if (Number.isNaN(valueNumber)) {
+    throw new Error(`The value for ${argument} is not an interger.`);
+  }
+
+  return valueNumber;
+
+}
```
Comment (pdehaan) [inline scripts/collect-unresolved-breaches-addresses.js:43]: We already include **arg** module in package.json, not sure if you just want to use that vs rolling your own arg parser (as pretty as it is).
https://github.com/mozilla/blurts-server/blob/915d9fc31e8de126afc35454f8720a83c15adfcb/package.json#L11

File: scripts/collect-unresolved-breaches-addresses.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  const limitQuery = getArgsValue("--limit");
+  const cohortSize = getArgsValue("--cohort-size");
+
+  console.log(`The limit of this query is ${limitQuery}`);
+  console.log(`The target cohort size of this query is ${cohortSize}`);
+
+  // "SELECT primary_email, primary_sha1 FROM subscribers WHERE signup_language LIKE 'en%' AND breaches_resolved IS NULL ORDER BY random();"
+
+  const results = await knex("subscribers").where("signup_language", "like", "en%").andWhere({breaches_resolved: null}).orderByRaw("RANDOM()").limit(limitQuery).select("primary_email", "primary_sha1");
```
Comment (pdehaan) [inline scripts/collect-unresolved-breaches-addresses.js:59]: suuuper nit: Move `.select("primary_email, "primary_sha1")` after the `knex("subscribers")` to look more SQLey? Also might benefit from a couple line breaks to make it easier to parse.

```js
const results = await knex("subscribers")
  .select("primary_email", "primary_sha1")
  .where("signup_language", "like", "en%")
  .andWhere({breaches_resolved: null})
  .orderByRaw("RANDOM()")
  .limit(limitQuery);
```

**Pull request merged**